lib/match.js

```diff
@@ -47,7 +47,7 @@ function matchCore(value, expected, partialModeFlag) {
  * @api private
  */
 function compareArrays(array, expected, partialMode) {
-  if (!partialMode && array.length !== expected.length) return false;
+  if (array.length !== expected.length) return false;
```

Comment:
I changed this since in the readme there is "Only objects (and not arrays) are affected by partial mode.", So I guess the array length should always be the same. Let me know if I should revert